### PR TITLE
Add oVirt ovirt_vms_facts module

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- ovirt_vms_facts
##### SUMMARY

This PR adds new _ovirt_vms_facts_ module, which handle facts about oVirt virtual machines.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
@mwperina @ryansb

File: cloud/ovirt/ovirt_vms_facts.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        - "For example to search VM X from cluster Y use following pattern:
+           name=X and cluster=Y"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all VMs named C<rhel**> from cluster C(west):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_vms_facts.py]: I'd rephrase to: 'Gather facts about all VMs which names start with C<centos> and belong to cluster C(west):'

File: cloud/ovirt/ovirt_vms_facts.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+- ovirt_vms_facts:
+    pattern: name=rhel* and cluster=west
```
Comment (mwperina) [inline cloud/ovirt/ovirt_vms_facts.py]: rhel -> centos

**Review by mwperina — CHANGES_REQUESTED** [review]

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! :+1:

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_vms_facts.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        module.exit_json(
+            changed=False,
+            ansible_facts=dict(
+                ovirt_vms=[
+                    get_dict_of_struct(c) for c in vms
+                ],
+            ),
+        )
+    except Exception as e:
+        module.fail_json(msg=str(e))
```
Comment (ryansb) [inline cloud/ovirt/ovirt_vms_facts.py]: Can you add the exception to the failure here, such as:

```
try:
    # stuff
except Exception as e:
    import traceback
    module.fail_json(msg=str(e), exception=traceback.format_exc())
```

Comment (machacekondra) [reply cloud/ovirt/ovirt_vms_facts.py]: done

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Review by mwperina — APPROVED** [review]
:+1:
